src/tracks/playabletrack/notetrack/ui/NoteTrackAffordanceControls.cpp

```diff
@@ -0,0 +1,83 @@
+#include "NoteTrackAffordanceControls.h"
+
+#include "../../../ui/AffordanceHandle.h"
+#include "../../../../AllThemeResources.h"
+#include "../../../../AColor.h"
+#include "../../../../NoteTrack.h"
+#include "../../../../ViewInfo.h"
+#include "../../../../TrackArtist.h"
+#include "../../../../TrackPanelMouseEvent.h"
+#include "../../../../TrackPanel.h"
+#include "../../../../TrackPanelDrawingContext.h"
+
+#include "../lib-src/header-substitutes/allegro.h"
+
+
+NoteTrackAffordanceControls::NoteTrackAffordanceControls(const std::shared_ptr<Track>& pTrack)
+    : CommonTrackCell(pTrack)
+{
+
+}
+
+std::vector<UIHandlePtr> NoteTrackAffordanceControls::HitTest(const TrackPanelMouseState& state, const AudacityProject* pProject)
+{
+    std::vector<UIHandlePtr> results;
+
+    auto track = FindTrack();
+    const auto nt = std::static_pointer_cast<const NoteTrack>(track->SubstitutePendingChangedTrack());
+
+    const auto rect = state.rect;
+
+    auto& trackPanel = TrackPanel::Get(*pProject);
+    auto& zoomInfo = *trackPanel.GetViewInfo();
+    auto left = zoomInfo.TimeToPosition(nt->GetOffset(), rect.x);
+    auto right = zoomInfo.TimeToPosition(nt->GetOffset() + nt->GetSeq().get_real_dur(), rect.x);
+    auto headerRect = wxRect(left, rect.y, right - left, rect.height);
+
+    auto px = state.state.m_x;
+    auto py = state.state.m_y;
+
+    if (px >= headerRect.GetLeft() && px <= headerRect.GetRight() &&
+        py >= headerRect.GetTop() && py <= headerRect.GetBottom())
+    {
+        results.push_back(AffordanceHandle::HitAnywhere(mAffordanceHandle, track));
+    }
+
+    return results;
+}
+
+void NoteTrackAffordanceControls::Draw(TrackPanelDrawingContext& context, const wxRect& rect, unsigned iPass)
+{
+    if (iPass == TrackArtist::PassBackground) {
+        const auto nt = std::static_pointer_cast<const NoteTrack>(FindTrack()->SubstitutePendingChangedTrack());
+        const auto artist = TrackArtist::Get(context);
+
+        TrackArt::DrawBackgroundWithSelection(context, rect, nt.get(), AColor::labelSelectedBrush, AColor::labelUnselectedBrush);
+
+        const auto& zoomInfo = *artist->pZoomInfo;
+        auto left = zoomInfo.TimeToPosition(nt->GetOffset(), rect.x);
+        auto right = zoomInfo.TimeToPosition(nt->GetOffset() + nt->GetSeq().get_real_dur(), rect.x);
+        auto clipRect = wxRect(left, rect.y, right - left + 1, rect.height);
+
+        auto px = context.lastState.m_x;
+        auto py = context.lastState.m_y;
+
+        auto selected = IsSelected();
+        auto highlight = selected ||
+            (px >= clipRect.GetLeft() && px <= clipRect.GetRight() &&
+                py >= clipRect.GetTop() && py <= clipRect.GetBottom());
+
+        context.dc.SetClippingRegion(rect);
+        TrackArt::DrawClipAffordance(context.dc, clipRect, highlight, selected);
+        context.dc.DestroyClippingRegion();
```

Comment:
I might want to make a RAII class, perhaps in TrackArtist.h, called TempClippingRegion which does SetClippingRegion in its ctor DestroyClipppingRegion in its door... Define it once, reuse the class in a few places.

Reply:
But then... perhaps the setting and destroying of the clipping region can happen inside DrawClipAffordance?  Because that function always requires it?  Then you don't need to repeat it for wave track affordance.

Reply:
DrawClipAffordance does not have enough of context to make proper clipping, also in example with wave clips, that would lead to redundant calls